File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        String three = "";
        int num = 3;
        int numCount = 1;
        while(n > 0){
            if(n >= num){
                num *= 3;
                numCount++;
                continue;
            }else{
                num /= 3;
                int cnt = n / num;
                three += cnt;
                n %= num;
            }
        }
        int len = three.length();
        if(len < numCount){
            for(int i = 0; i < numCount - len; i++){
                three += "0";
            }
        }

        for(int i = 0; i < three.length(); i++){
            String s = Character.toString(three.charAt(i));
            int res = Integer.parseInt(s);
            if(i == 0){
                answer += res;
            }else{
                answer += res * Math.pow(3, i);
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.swift
import Foundation

func solution(_ n:Int) -> Int {
    var result = 0
    var three: String = ""
    var num = 3
    var numCount = 1
    var N = n
    
    while (N > 0) {
        if (N >= num) {
            num *= 3
            numCount += 1
            continue
        } else {
            num /= 3
            var cnt = N / num
            three += String(cnt)
            N = N % num
        }
    }
    if three.count < numCount {
        for i in 0 ..< numCount - three.count {
            three += "0"
        }
    }
    var cnt = 0
    for i in three {
        if let res = Int(String(i)) {
            if cnt == 0 {
                result += res
            } else {
                result += Int(Double(res) * pow(Double(3), Double(cnt)))
            }
        }
        cnt += 1
    }
    return result
}